tests/unit/test_session.py

```diff
@@ -122,6 +122,69 @@ def test_close_exception():
         session.close()
 
 
+def test_close_session_in_stored_procedure_no_op():
+    fake_connection = mock.create_autospec(ServerConnection)
+    fake_connection._conn = mock.Mock()
+    fake_connection.is_closed = MagicMock(return_value=False)
+    session = Session(fake_connection)
+    with mock.patch.object(
+        snowflake.snowpark.session, "is_in_stored_procedure"
+    ) as mock_fn, mock.patch.object(
+        session._conn, "close"
+    ) as mock_close, mock.patch.object(
+        session, "cancel_all"
+    ) as mock_cancel_all, mock.patch.object(
+        snowflake.snowpark.session, "_remove_session"
+    ) as mock_remove:
+        mock_fn.return_value = True
+        session.close()
+        mock_cancel_all.assert_not_called()
+        mock_close.assert_not_called()
+        mock_remove.assert_not_called()
+
+
+def test_close_session_in_stored_procedure_log_level_warning(caplog):
+    caplog.set_level(logging.WARNING)
+    fake_connection = mock.create_autospec(ServerConnection)
+    fake_connection._conn = mock.Mock()
+    fake_connection.is_closed = MagicMock(return_value=False)
+    session = Session(fake_connection)
+    with mock.patch.object(
+        snowflake.snowpark.session, "is_in_stored_procedure"
+    ) as mock_fn:
+        mock_fn.return_value = True
+        session.close()
+    assert "Closing a session in a stored procedure is a no-op." in caplog.text
```

Comment:
the three tests here look similar. Can you use `@pytest.mark.parametrize` to test this.

Reply:
Great suggestion 👍
Combined the three log level tests into one using pytest parametrize.